src/runtime/swg-client-event-manager.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import {isObject, isFunction, isEnumValue} from '../utils/types';
+import * as EventManagerApi from '../api/swg-client-event-manager-api';
+
+/**
+ * Helper function to describe an issue with an event object
+ * @param {!string} valueName
+ * @param {?*} value
+ * @returns {!string}
+ */
+function createEventErrorMessage(valueName, value) {
+  return 'An SwgClientEvent has an invalid ' + valueName + '(' + value + ')';
+}
+
+/**
+ * Throws an error if the event is invalid.
+ * @param {!EventManagerApi.SwgClientEvent} event
+ * @returns {!Promise}
+ */
+function validateEvent(event) {
+  if (!event) {
```

Comment:
by virtue of adding  a ! to the argument in line 33, this check will never happen. you can remove.

Reply:
My test code passes a null in and generates the error I created.  I replaced it with an isObject check.